test/tape/parse-connection.js

```diff
@@ -49,6 +51,92 @@ test('mysql connection protocol with query string params', function (t) {
         port: '3306',
         database: 'testdb',
         foo: 'bar',
+        anotherParam: '/a/b/c',
+      },
+    }
+  );
+});
+
+test('mysql connection protocol allows skip username', function(t) {
+  t.plan(1);
+  t.deepEqual(
+    parseConnection(
+      'mysql://:pass@path.to.some-url:3306/testdb?foo=bar&anotherParam=%2Fa%2Fb%2Fc'
+    ),
+    {
+      client: 'mysql',
+      connection: {
+        user: '',
+        password: 'pass',
+        host: 'path.to.some-url',
+        port: '3306',
+        database: 'testdb',
+        foo: 'bar',
+        anotherParam: '/a/b/c',
+      },
+    }
+  );
+});
+
+test('mysql connection protocol allows skip password', function(t) {
+  t.plan(1);
+  t.deepEqual(
+    parseConnection('mysql://username@path.to.some-url:3306/testdb'),
+    {
+      client: 'mysql',
+      connection: {
+        user: 'username',
+        host: 'path.to.some-url',
+        port: '3306',
+        database: 'testdb',
+      },
+    }
+  );
+});
+
+test('mysql connection protocol allows empty password', function(t) {
+  t.plan(1);
+  t.deepEqual(
+    parseConnection('mysql://username:@path.to.some-url:3306/testdb'),
+    {
+      client: 'mysql',
+      connection: {
+        user: 'username',
+        host: 'path.to.some-url',
+        port: '3306',
+        database: 'testdb',
+      },
+    }
+  );
+});
+
+test('decodes username and password', function(t) {
```

Comment:
we also need a test for sqlite Windows path, e. g. 'C:/node/database.sqlite'

Reply:
done